nservicebus/azure-service-bus/sanitization.md

```diff
@@ -43,24 +47,43 @@ To perform manual sanitization, inspect entity name for invalid characters and n
 
 All entities treated the same for allowed characters. Only letters, numbers, periods (`.`), hyphens (`-`), and underscores (`_`) were allowed. Maximum length was predefined by the transport.
 
-In Version 6.4.x [Naming Conventions](/nservicebus/azure-service-bus/native-integration.md) were added to allow custom sanitization of queues, topics, and subscriptions.
+In Version 6.4.x [Naming Conventions](/nservicebus/azure-service-bus/naming-conventions.md) were added to allow custom sanitization of queues, topics, and subscriptions.
 
 
 ### Versions 7 and above
 
-By default, the transport does not sanitize entities. Instead, it passes entity paths/names as-is to the broker.
+By default, the transport uses `ThrowOnFailedValidation` sanitization strategy. This sanitization validates entity path/name and if it's valid, passes as-is to the broker. For an invalid entity path/name, an exception is thrown. Validation rules can be adjusted by providing custom validation rules per entity type. 
+
+snippet: asb-ThrowOnFailedValidation-sanitization-overrides
+
+Where `ValidationResult` provides the following
+ * Characters are valid or not
+ * Length is valid or not
+
+To customize sanitization for some of the entities, `ValidateAndHashIfNeeded` strategy can be used. This strategy allows to specify sanitization rules to remove invalid characters and hashing algorithm to shorten entity path/name that is exceeding maximum length.
+
+NOTE: `ValidateAndHashIfNeeded` is using validation rules to determine what needs to be sanitized. First step, invalid characters are removed. Second step, hashing is applied if length is still exceeding the maximum allowed length.
 
-In cases where entities are too long, and there is no way to shorten those, custom sanitization can be applied on the transport level to ensure names adhere to the Azure Service Bus service rules.
+snippet: asb-ValidateAndHashIfNeeded-sanitization-overrides
+
+In cases an alternative sanitization strategy is required, a custom sanitization can be applied.
 
 snippet: asb-custom-sanitization
 
 Custom sanitization strategy definition:
 
 snippet: asb-custom-sanitization-strategy
 
-When using `EndpointOrientedTopology`, the transport should be configured to use `EndpointOrientedTopologySanitization` sanitization strategy for backward compatibility with endpoints versions 6 and below.
+If the implementation of a sanitization strategy requires configuration settings, these settings can be accessed by letting NServiceBus inject the `ReadOnlySettings` into the constructor of the sanitization strategy.
```

Comment:
technically it could be property as well.

> If the implementation of a sanitization strategy requires configuration settings, these settings can be accessed using [Dependency Injection](/nservicebus/containers/) to access an instance of `ReadOnlySettings`.